<launch>
  <arg name="robot" default="scitos"/>

  <!-- views: DEP or IND -->
  <arg name="setting" default="DEP"/>  
  <!-- algorithm: SRC, DRC, SR, DR, SC, DC --> 
  <arg name="alg" default="SRC"/> 
  
  <arg name="num_of_plans" default="1000"/>
  <!-- <arg name="plan_length" default="25"/> -->
  <arg name="time_window" default="10"/>
  <arg name="rho" default="1.0"/>



  <arg name="input_views" default="$(find viper)/data/op/$(arg robot)/view_keys.json"/>
  <arg name="input_file" default="$(find viper)/data/op/$(arg robot)/plans/View$(arg setting)-Alg$(arg alg)-N$(arg num_of_plans)-T$(arg time_window)-Rho$(arg rho).json"/>
  <!-- <arg name="input_file" default="$(find viper)/data/$(arg robot)/plans.json"/> -->
  <arg name="input_file_values" default="$(find viper)/data/op/$(arg robot)/view_values.json"/>
  <arg name="input_file_costs" default="$(find viper)/data/op/$(arg robot)/view_costs.json"/>
  <arg name="output_file" default="$(find viper)/data/op/$(arg robot)/plans/View$(arg setting)-Alg$(arg alg)-N$(arg num_of_plans)-T$(arg time_window)-Rho$(arg rho)-BEST.json"/>
  <!-- <arg name="output_file" default="$(find viper)/data/$(arg robot)/plan_values.json"/>-->

  <include file="$(find viper)/launch/$(arg robot)/_evaluate_plans.launch"/>

  <node name="evaluate_plans" pkg="viper" type="eval_plans.py" output="screen">
    <param name="robot" type="string" value="$(arg robot)"/>
    <param name="input_views" type="string" value="$(arg input_views)"/>
    <param name="input_file" type="string" value="$(arg input_file)"/>
    <param name="input_file_values" type="string" value="$(arg input_file_values)"/>
    <param name="input_file_costs"  type="string" value="$(arg input_file_costs)"/>
    <param name="output_file" type="string" value="$(arg output_file)"/>
  </node>
</launch>
